kernel/src/process_manager/allocation.rs

```diff
@@ -7,39 +7,84 @@ use crate::cpu::control_regs::read_cr3;
 use crate::{paddr_as_slice, map_paddr, vaddr_as_slice};
 use crate::mm::PerCPUPageMappingGuard;
 
+const ALLOCATION_VADDR_START: u64 = 0x30000000000u64;
+pub const DEFAULT_ALLOCATION_RANGE_MOUNT: usize = 6;
+const PGD_SHIFT: u64 = 39;
 #[repr(C)]
-#[derive(Debug, Clone, Copy)]
+#[derive(Debug, Clone, Copy, Default)]
 pub struct AllocationRange(pub u64, pub u64);
 
- impl AllocationRange {
+impl AllocationRange {
 
     pub fn allocate(&mut self, pages: u64){
-        // Reuses the Process page managment to add new memory to the Monitor
         let mut page_table_ref = ProcessPageTableRef::default();
         page_table_ref.set_external_table(read_cr3().bits() as u64);
+        self.allocate_(&mut page_table_ref, pages, ALLOCATION_VADDR_START, true);
+    }
+
+    pub fn allocate_with_start_addr(&mut self, page_table_ref: &mut ProcessPageTableRef, pages: u64, start_addr: u64){
+        self.allocate_(page_table_ref, pages, start_addr, false);
+    }
+
+    fn allocate_(&mut self, page_table_ref: &mut ProcessPageTableRef, pages: u64, start_addr: u64, mount: bool){
+        // Reuses the Process page managment to add new memory to the Monitor
+        //let mut page_table_ref = ProcessPageTableRef::default();
+        //page_table_ref.set_external_table(read_cr3().bits() as u64);
         let table_flags = ProcessPageFlags::PRESENT | ProcessPageFlags::WRITABLE |
-                          ProcessPageFlags::DIRTY | ProcessPageFlags::ACCESSED;
+        ProcessPageFlags::DIRTY | ProcessPageFlags::ACCESSED;
 
-        let start_address = VirtAddr::from(0x30000000000u64);
+        let start_address = VirtAddr::from(start_addr);
 
         for i in 0..(pages as usize) {
             let current_page = allocate_page();
```

Comment:
I think for non-mount case (i.e., adding pages for trustlets), rmpadjust is needed
```c
            if !mount {
                let (mapping, _page_mapped) = paddr_as_slice!(current_page);
                rmp_adjust(mapping.virt_addr(), RMPFlags::VMPL1 | RMPFlags::RWX , PageSize::Regular).unwra
p();
            }
```